cms/datasets/models.py

```diff
@@ -1,58 +1,150 @@
+        # For detail responses, we may need to adjust the structure.
+        # This only applies to the detail URL which currently uses the old format.
+        if response.get("current"):
+            # If it's a versioned dataset response using the old format, convert to new format
+            dataset_data = convert_old_dataset_format(response["current"])
+            # Store the next version info if available (this becomes our unpublished version)
+            next_entry = response.get("next")
+            dataset_data["next"] = convert_old_dataset_format(next_entry) if next_entry else {}
+            return dataset_data
+        return response
+
 
 class ONSDataset(APIModel):
     base_query_class = ONSDatasetApiQuerySet
 
     search_fields: ClassVar[list[str]] = ["title", "version", "formatted_edition"]
 
     class Meta:
-        base_url: str = settings.DATASETS_API_BASE_URL
+        base_url: str = settings.DATASETS_API_EDITIONS_URL
         detail_url: str = f"{settings.DATASETS_API_BASE_URL}/%s"
-        fields: ClassVar = ["id", "description", "title", "version", "edition"]
+        fields: ClassVar = ["id", "description", "title", "version", "edition", "published", "next"]
         pagination_style = "offset-limit"
         verbose_name_plural = "ONS Datasets"
 
     @classmethod
     def from_query_data(cls, data: Mapping) -> "ONSDataset":
-        url: str = data["links"]["latest_version"]["href"]
-        edition_match = EDITIONS_PATTERN.search(url)
-        if not edition_match:
-            raise ValueError(f"Found invalid dataset URL, missing edition: {url} for dataset: {data['id']}")
-        edition = edition_match.group(1)
+        # Handle new /v1/dataset-editions response structure
+        dataset_id = data.get("dataset_id", data.get("id", ""))
+        title = data.get("title") or "Title not provided"
+        description = data.get("description") or "Description not provided"
+        edition = data.get("edition", "")
+        published = data.get("published", True)
+        # Handle next (unpublished) version if present
+        next_dataset_entry = data.get("next")
+
+        if next_dataset_entry:
+            # Recursively create ONSDataset for the unpublished version
+            next_dataset_entry = ONSDataset.from_query_data(next_dataset_entry)
+
+        # Extract version from latest_version object
+        latest_version = data.get("latest_version", {})
+        version_id = latest_version.get("id", "1") if isinstance(latest_version, dict) else "1"
+
         return cls(
-            id=data["id"],
-            title=data["title"],
-            description=data["description"],
-            version=data["links"]["latest_version"]["id"],
+            id=dataset_id,
+            title=title,
+            description=description,
+            version=version_id,
             edition=edition,
+            published=published,
+            next=next_dataset_entry,
         )
```

Comment:
Im not following why we store the next dataset entry against the dataset. It should always be one of the other. `next` if we are fetching unpublished, otherwise `current` / top level.

Reply:
I suggest adding a comment explaining that's why we do this. That because we can't know at this stage whether we are looking for published or unpublished, we return both and choose the correct one later in the request cycle.